### DISCUSSION TITLE
Anyone having CSS problems after updating?

### DISCUSSION THREAD

**Discussion by PetersOtto** [datenstrom/community#969] [Ask a question] [answered]
Hello everyone!

I have tried to bring yellow up to date. I have installed the update with the edit extension. I could not update directly to yellow 0.9. Therefore I did the update in the following two steps.

Step 1:
![240408_yellow_update_step1](https://github.com/datenstrom/yellow/assets/52236754/ea69ed26-d4fd-4973-9d9b-3a745a70ff34)

Step 2:
![240408_yellow_update_step2](https://github.com/datenstrom/yellow/assets/52236754/08662b7e-42a7-4b7a-874c-9dcf9ff86295)

Already after the first step the `css` and `js` files were not found (error 404). But the links/paths to it are correct.

I use my own theme. This theme has the following file structure.

```
├── content               
│   ├── 1-home            
│   ├── 9-about           
│   └── shared            
├── media                 
│   ├── downloads         
│   ├── images            
│   └── thumbnails        
└── system                
    ├── extensions
    │   ├── myTheme
    │   │   ├── css
    │   │   ├── favicon
    │   │   ├── fonts
    │   │   ├── js
    │   │   ├── scss    
    ├── layouts           
    ├── themes
    │   ├── myTheme.css (empty)     
    └── trash 
``` 

Does yellow perhaps lock `css` and `js` files in subfolders of extensions?

**Comment by markseuffert** [comment, marked as answer]
I think I know what happened. Direct access to the system folder is not allowed, so the [core](https://github.com/annaesvensson/yellow-core) maps system files via a virtual location. This virtual mapping has changed in the new release and it could have broken your theme. Can you try the following and move the `myTheme` folder from `system/extensions/myTheme` to `system/themes/myTheme`. Possibly adjust the virtual location in your HTML files. It's now the same virtual location for layouts/themes/workers. Here is an example folder structure:

```
├── content               
├── media                 
└── system                  <- direct access to the system folder is not allowed
    ├── extensions
    ├── layouts
    ├── themes
    │   └── myTheme         <- this folder is mapped to virtual location /assets/myTheme/
    │       ├── css
    │       ├── favicon
    │       ├── fonts
    │       ├── js
    │       └── scss 
    └── workers
```

Sorry about the trouble with the update to the new release. Does this work for you?

**Comment by PetersOtto** [comment]
Yes, it works for me. It's a bit more work for me now (more than one website), but I wanted to change it anyway.
Thanks for your help!

**Comment by glow12121** [comment]
```
│   └── myTheme         <- this folder is mapped to virtual location /assets/myTheme/
```

Did you imply creating a separate folder for your theme? If so, this leads to errors during `build`:

```
$ php yellow.php build
ERROR building location '/', HTTP/1.1 500 Server error: Theme 'smalltown' does not exist!
ERROR building location '/page1', HTTP/1.1 500 Server error: Theme 'smalltown' does not exist!
ERROR building location '/error/', HTTP/1.1 500 Server error: Theme 'smalltown' does not exist!
Building static website 100%... done
Yellow build: 8 files, 6 errors
```

**Reply by annaesvensson** [reply]
You can have subfolders in the theme folder, for more information see #928.
